### ISSUE TITLE
How to build a headless executable CLI?

### ISSUE THREAD

**Issue by msafi** [spesmilo/electrum#6166]
I want to use Electrum CLI to provide wallet functionality in my app. So basically I want to build a GUI on top of an Electrum CLI. So I was wondering if it's possible to build executable Electrum CLIs for Windows, macOS, and Linux? And how to do it?

**Comment by ecdsa** [comment]
you do not need to 'build' it, just run Electrum from python sources.
if you use only the CLI you do not need to install Qt

**Comment by msafi** [comment]
@ecdsa I'm not sure what you mean by "just run Electrum from python sources".

I want to ship Electrum CLI as part of a TypeScript Electron desktop app. So I need an executable binary of Electrum's CLI.

I tried to create this executable binary on my macOS machine, but I don't think I did it right. I cloned the `spesmilo/electrum`, ran `brew install qt5 pyqt5 gettext`, then ran `python3 run_electrum` and got this:

```bash
Error: at least one of ('pycryptodomex', 'cryptography') needs to be installed.
```

But even if I fix this error and manage to run Electrum with the `run_electrum` script, I don't think that alone gets me to my goal of having a portable Electrum CLI, right?

**Comment by SomberNight** [comment]
I don't know how you could have a "portable" binary between e.g. Windows and MacOS.
The typical solution is just to build one for each platform.
This is also what we do when distributing Electrum.

You could build Electrum using the [build scripts in this repository](https://github.com/spesmilo/electrum/blob/b9f20d2c79c1c27fb11aff296ec63b78683a8c38/README.rst#creating-binaries), separate binary for each target platform. Those binaries can be used for CLI (well, Windows is problematic atm, see https://github.com/spesmilo/electrum/issues/2592). If you want, you can strip out Qt by customising the build scripts.

**Comment by msafi** [comment]
@SomberNight Yes, sorry I wasn't clear. I do expect to build a separate binary for each platform.

It's a bummer that the CLI doesn't work on Windows though!

**Comment by Dibbyo456** [comment]
Do anyone knows how to build electrum cli only? I don't want GUI because it will be running on a server.
